src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1771,6 +1771,13 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, SyntaxToken tar
 #endif
 
             var enclosingSymbol = semanticModel.GetEnclosingSymbol(targetToken.SpanStart, cancellationToken);
+
+            while (enclosingSymbol is IMethodSymbol method && method.MethodKind == MethodKind.LocalFunction)
+            {
+                // It is allowed to reference the instance (`this`) within a local function, as long as the containing method allows it
+                enclosingSymbol = method.ContainingSymbol;
```

Comment:
i find this fix surprising.  Is a local function treated as static (even though it can use 'this')?

Reply:
Tagging @agocke @gafter to confirm expected semantic model behavior on local functions (they currently have `IsStatic = true` even if the enclosing method isn't static).

Reply:
Sounds wrong to me. If they're emitted as static methods, that's just a codegen decision.

Reply:
From discussion with @agocke and @AlekseyTs, there are two bugs:
- the compiler `LocalFunctionSymbol.IsStatic` should always return `false`, but currently always returns `true`.
- given that, the IDE should not consider `LocalFunctionSymbol.IsStatic` to decide whether `this` can be used in the current context.

Given that, I will revert this PR to my original fix.

Reply:
> the compiler LocalFunctionSymbol.IsStatic should always return false, but currently always returns true.

Not disagreeing... but can an explanation be provided as to why?

This seems similar to props+accessors.  Accessors can't have static/non-static info on them.  They just 'inherit' from the containing prop.  Seems similar to local-functions/containers.  And it would seem to make the most sense.  It would also tie into one of the main concepts of Static/Instance (namely, whether or not you can use 'this'..).  Thoughts?

Reply:
It looks like there is some [discussion](https://github.com/dotnet/roslyn/issues/27719#issuecomment-396363477) going on for the compiler side (maybe it should always be `true` after all) ;-)